src/tests/history.rs

```diff
@@ -228,7 +228,7 @@ fn time_barrier() {
 fn generate_history_lines(item_count: usize, idx: usize) -> Vec<WString> {
     let mut result = Vec::with_capacity(item_count);
     for i in 0..item_count {
-        result.push(sprintf!("%lu %lu", idx, i));
+        result.push(sprintf!("%u %u", idx, i));
```

Comment:
Not to beat the dead horse but we [should move to Rust format string syntax](https://github.com/fish-shell/fish-shell/pull/11857#discussion_r2404109075),
ideally with all the compile-time checks that we don't have with `sprintf!`.
I guess this PR is a step in that direction, it will make the conversion step easier.